### DISCUSSION TITLE
Add PostgreSQL custom schema support

### DISCUSSION THREAD

**Discussion by zvrh** [dani-garcia/vaultwarden#2208] [Ideas]
Currently, when using PostgreSQL database, the public schema is used by default and cannot be changed. Can you provide the function of customizing the schema name?

I tried to use the search_path parameter in DATABASE_URL, but it didn't work properly.

```
postgresql://user_name:user_password@db_host:5432/db_name?search_path=db_schema
```

**Comment by BlackDex** [comment]
Im not sure of this will work, but you could try this.
```ini
DATABASE_URL="postgresql://user_name:user_password@db_host:5432/db_name?application_name=vaultwarden&options=-c search_path%3Ddb_schema"
```
Note the `%3D` which is an url-encoded `=` sign.

Please let us know is this works.

**Reply by zvrh** [reply]
Thanks, this works. But I need to remove the double quotes when I use it in docker compose, otherwise an error will be reported.
``` yml
environment:
      - DATABASE_URL=postgresql://vaultwarden:xxxxxx:@postgres:5432/vaultwarden?application_name=vaultwarden&options=-c search_path%3Dvault
```
